api/hasura/event-triggers/refund-pending-gift.ts

```diff
@@ -0,0 +1,95 @@
+import assert from 'assert';
+
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+import { gql } from '../../../api-lib/Gql';
+import { EventTriggerPayload } from '../../../api-lib/types';
+import { verifyHasuraRequestMiddleware } from '../../../api-lib/validate';
+
+async function handler(req: VercelRequest, res: VercelResponse) {
+  // no parsing should be needed here since this data comes straight from
+  // the database and zeus keeps this consistent for us
+  const payload: EventTriggerPayload<
+    'pending_token_gifts',
+    'DELETE' | 'UPDATE'
+  > = req.body;
+  const {
+    recipient_id,
+    sender_id,
+    tokens: tokensToRefund,
+    circle_id,
+  } = payload.event.data.old;
+
+  const newTokenAmount = payload.event.data.new?.tokens ?? 0;
+
+  if (tokensToRefund === 0 || newTokenAmount != 0) {
+    res.status(200).json({
+      message: `Not a refund event.`,
+    });
+    return;
+  }
+
+  try {
+    const { users_by_pk: sender } = await gql.q('query')({
+      users_by_pk: [
+        { id: sender_id },
+        {
+          non_giver: true,
+          give_token_remaining: true,
+          name: true,
+        },
+      ],
+    });
+    const { users_by_pk: recipient } = await gql.q('query')({
+      users_by_pk: [
+        { id: recipient_id },
+        {
+          non_receiver: true,
+          give_token_remaining: true,
+          name: true,
+        },
+      ],
+    });
+    assert(sender, `dangling give without user: ${sender_id}`);
+    assert(recipient, `dangling give without user: ${recipient_id}`);
+    const {
+      give_token_remaining: originalGive,
+      name: senderName,
+      non_giver,
+    } = sender;
+
+    const { name: recipientName, non_receiver } = recipient;
+
+    if (tokensToRefund > 0 && (non_giver || non_receiver)) {
```

Comment:
just missing an mutation update  to deduct `give_token_received` from the recipient when switching to `non_giver` of the sender